LLMonFHIR/FHIRInterpretation/UserStudy/UserStudyChatView.swift

```diff
@@ -219,44 +39,33 @@ struct UserStudyChatView: View {
 
     @ViewBuilder private var chatContent: some View {
         ChatView(
-            Binding(
-                get: {
-                    var chat = interpreter.llm.context.chat
-                    if viewModel.navigationState == .completed {
-                        let surveyReport = viewModel.generateReport()
-                        chat.append(ChatEntity(role: .hidden(type: .init(name: "SURVEY_REPORT")), content: surveyReport))
-                    }
-                    return chat
-                },
-                set: { interpreter.llm.context.chat = $0 }
-            ),
-            disableInput: isInputDisabled,
+            viewModel.chatBinding,
+            disableInput: viewModel.isProcessing,
             speechToText: false,
             exportFormat: viewModel.navigationState == .completed ? .text : nil,
-            messagePendingAnimation: .manual(shouldDisplay: interpreter.viewState == .processing || noAssistantMessages)
+            messagePendingAnimation: .manual(shouldDisplay: viewModel.isProcessing)
         )
-            .viewStateAlert(state: interpreter.llm.state)
-            .onChange(of: interpreter.llm.context, initial: true) {
-                if interpreter.llm.state != .generating && interpreter.llm.context.chat.last?.role == .user {
-                    interpreter.queryLLM()
-                }
-            }
-            .onAppear {
-                guard noUserMessages else {
-                    return
-                }
-                interpreter.resetChat()
-                interpreter.queryLLM()
+            .viewStateAlert(state: viewModel.llmSession.state)
+            .onChange(of: viewModel.llmSession.context, initial: true) {
+                viewModel.generateAssistantResponse()
             }
     }
 
-
     /// Creates a new user study chat view
+    ///
+    /// This initializer sets up a view for conducting a structured study with
+    /// chat-based interactions and survey tasks.
+    ///
     /// - Parameter survey: The survey configuration that defines the study's tasks and structure
-    init(survey: Survey) {
-        _viewModel = StateObject(wrappedValue: UserStudyViewModel(survey: survey))
+    init(survey: Survey, interpreter: FHIRMultipleResourceInterpreter) {
```

Comment:
Wondering if there might be any advantage of passing the interpreter within the environment?

Reply:
I don’t see any advantage either. I’d be happy to do this in a follow-up PR. I just didn’t want to introduce too many changes into this one PR.

Reply:
Ok